File: src/Managers/FileManager.cpp
#include<iostream>
#include<fstream>

#include"Managers/FileManager.h"

using std::cout;
using std::endl;
using std::ifstream;
using std::ofstream;
using std::string;

namespace Managers
{
    FileManager::FileManager() {}
    FileManager::FileManager(string filePath)
    {
        this->filePath = filePath;
        readFile();
    }   


    void FileManager::saveFile(string newFilePath)
    {
        if (!fileRead)
            readFile();

        ofstream of{newFilePath, ofstream::binary};
        of.write(fileBuffer, fileBufferLength);
        of.close();
    }

    void FileManager::readFile()
    {
        ifstream is{filePath, ifstream::binary};
        if (is)
        {
            is.seekg (0, is.end);
            fileBufferLength = is.tellg();
            is.seekg (0, is.beg);

            fileBuffer = new char [fileBufferLength];

            cout<< "Reading "<<fileBufferLength<<" characters... "<<endl;;
            is.read (fileBuffer,fileBufferLength);

            if (is)
                cout<<"all characters read successfully.";
            else
                cout<<"error: only "<<is.gcount()<<" could be read";
            cout<<endl;
            is.close();
            fileRead = true;
        }
    }
    void FileManager::modifyFilePath(string filePath)
    {
        this->filePath = filePath;
    }

    char* FileManager::retrieveFileBuffer() const
    {
        return fileBuffer;
    }

    int FileManager::retrieveFileBufferLength() const { return fileBufferLength; }
}

File: src/managers/file_manager.rs
use std::default::Default;


pub struct FileManager {
    pub filepath: String,
    pub filebuffer: String,
    pub file_read: bool,
    pub file_buffer_length: i32,
}

impl Default for FileManager {
    fn default() -> Self {
        FileManager {
            filepath: String::new(),
            filebuffer: String::new(),
            file_read: false,
            file_buffer_length: -1
        }
    }
}


impl FileManager {
// TODO: Implement
pub fn init(&mut self) {
    self.read_file()
}
    pub fn save_file(&mut self, filepath: &String) {
        if !self.file_read {
            self.read_file();
        }

            let mut file = File::open(filepath)?;
    let mut buffer = Vec::new();
    file.read_to_end(&mut buffer)?;
        self.file_buffer_length = buffer.len();
    self.filebuffer = String::from_utf8(buffer); // Assuming UTF-8 encoding
    }
    pub fn modify_file_path(&mut self, file: &String) {
        self.filepath = file;
    }

    pub fn retrieve_file_buffer(&self) -> String {
        self.filebuffer;
    }

    pub fn retrieve_file_length(&self) -> i32 {
        self.file_buffer_length;
    }

    fn read_file(&mut self) {
        let mut file = File::open(self.filepath)?;
        let mut buffer = Vec::new();
        file.read_to_end(&mut buffer)?;
        self.file_buffer_length = buffer.len();
        self.filebuffer = String::from_utf8(buffer); // Assuming UTF-8 encoding
        self.file_read = true;
    }
}